lib/tasks/valid_test_data_generators/applications_populater.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+require "tasks/valid_test_data_generators/school_urn_generator"
+require "tasks/valid_test_data_generators/trn_generator"
+
+module ValidTestDataGenerators
+  class ApplicationsPopulater
+    class << self
+      def populate(lead_provider:, cohort:, total_schools: 10, participants_per_school: 50)
+        new(lead_provider:, cohort:, participants_per_school:).populate(total_schools:)
+      end
+    end
+
+    def populate(total_schools: 10)
+      return unless Rails.env.in?(%w[development separation])
+
+      logger.info "ApplicationsPopulater: Started!"
+
+      ActiveRecord::Base.transaction do
+        generate_new_schools!(count: total_schools)
+      end
+
+      logger.info "ApplicationsPopulater: Finished!"
+    end
+
+  private
+
+    attr_reader :lead_provider, :cohort, :participants_per_school, :courses, :logger
+
+    def initialize(lead_provider:, cohort:, participants_per_school:, logger: Rails.logger)
+      @lead_provider = lead_provider
+      @cohort = cohort
+      @participants_per_school = participants_per_school
+      @logger = logger
+      @courses = Course.all.reject { |c| c.identifier == "npq-early-headship-coaching-offer" }
+    end
+
+    def generate_new_schools!(count:)
+      count.times { create_school_with(urn: SchoolURNGenerator.next) }
+    end
+
+    def find_or_create_participants(school:, number_of_participants:)
+      generate_new_participants(school:, count: number_of_participants)
+    end
+
+    def generate_new_participants(school:, count:)
+      count.times do
+        create_participant(school:)
+      end
+    end
+
+    def create_participant(school:)
+      name = Faker::Name.name
+      trn = TRNGenerator.next
+      course = courses.sample
+
+      user = User.create!(
+        full_name: name,
+        email: Faker::Internet.email(name:),
+        date_of_birth: Date.new(1990, 1, 1),
+        trn:,
+        trn_verified: true,
+        trn_auto_verified: true,
+        ecf_id: SecureRandom.uuid,
+        get_an_identity_id: SecureRandom.uuid,
+      )
+
+      application = Application.create!(
```

Comment:
discussed in person: thinking about using factories vs model itself, we can compare notes after